File: Java/2483. Minimum Penalty for a Shop.java
class Solution {
    public int bestClosingTime(String customers) {
        int cur = 0, min = 0, ans = 0;

        for (int i = 0; i < customers.length(); i++) {
            if (customers.charAt(i) == 'Y') {
                cur--;
            } else {
                cur++;
            }

            if (cur < min) {
                ans = i + 1;
                min = cur;
            }
        }

        return ans;
    }
}

File: Python3/2483. Minimum Penalty for a Shop.py
class Solution:
    def bestClosingTime(self, customers: str) -> int:
        cur = m = ans = 0

        for i, c in enumerate(customers):
            if c == 'Y':
                cur -= 1
            else:
                cur += 1

            if cur < m:
                ans = i + 1
                m = cur

        return ans